src/component/TodoList.tsx

```diff
@@ -0,0 +1,45 @@
+import { useContext } from 'react';
+import { TodoItem } from './TodoItem';
+import { TodosContext } from '../TodosContext';
+
+export default function TodoList() {
+  const {
+    todos,
+    setTodos,
+    isToggleAllStatus,
+    setIsToggleAllStatus,
+    todosFilter,
+  } = useContext(TodosContext);
+
+  const handleToggleAll = () => {
+    setTodos(todos.map(todo => ({
+      ...todo,
+      completed: !isToggleAllStatus,
+    })));
+    setIsToggleAllStatus(!isToggleAllStatus);
+  };
+
+  return (
+    <>
+      {!!todos.length && (
+        <section className="main">
+          <input
+            checked={isToggleAllStatus}
+            type="checkbox"
+            id="toggle-all"
+            className="toggle-all"
+            data-cy="toggleAll"
+            onChange={handleToggleAll}
+          />
+          <label htmlFor="toggle-all">Mark all as complete</label>
+
+          <ul className="todo-list" data-cy="todoList">
+            {todosFilter().map(todo => (
```

Comment:
please don't call a function from the template because it will be executed every time when the state change, so make sense to move it to the variable